tests/unittests/GradCheckTest.cpp

```diff
@@ -168,36 +168,135 @@ void performGradCheck(ExecutionEngine &EE, PlaceholderBindings &bindings,
+  Tensor *gradVarTensor = bindings.get(gradVar);
+  gradVarTensor->zero();
+
+  // Train the network just once to record the values of gradient for inputVar.
+  EE.run(bindings);
+
+  return gradVarTensor;
+}
+
+TEST_P(InterpreterGrad, gradientCheckConcatLeft) {
   PlaceholderBindings bindings;
-  size_t numOutputElem = 20;
+  size_t numOutputElem = 4;
 
   auto &mod = EE_.getModule();
   Function *F = mod.createFunction("main");
 
   auto *A = mod.createPlaceholder(ElemKind::FloatTy, {1, numOutputElem / 2},
                                   "A", false);
+  bindings.allocate(A)->getHandle() = {1, 2};
   auto *B = mod.createPlaceholder(ElemKind::FloatTy, {1, numOutputElem / 2},
                                   "B", false);
-  bindings.allocate(B)->zero();
+  bindings.allocate(B)->getHandle() = {3, 4};
 
-  auto *Exp = mod.createPlaceholder(ElemKind::FloatTy, {1, numOutputElem},
-                                    "exp", false);
   Node *O = F->createConcat("concat", {A, B}, 1);
-  O = F->createRegression("reg", O, Exp);
-  auto *result = F->createSave("ret", O);
+  Tensor expected(ElemKind::FloatTy, {1, numOutputElem / 2});
+  expected.getHandle() = {1, 2};
+  EXPECT_TRUE(expected.isEqual(*computeGradient(EE_, bindings, A, O)));
+}
 
-  Tensor inputs(ElemKind::FloatTy, {{1, numOutputElem / 2}});
-  Tensor outputs(ElemKind::FloatTy, {{1, numOutputElem}});
+TEST_P(InterpreterGrad, gradientCheckConcatRight) {
+  PlaceholderBindings bindings;
+  size_t numOutputElem = 4;
 
-  auto inputsH = inputs.getHandle<>();
-  auto outputsH = outputs.getHandle<>();
+  auto &mod = EE_.getModule();
+  Function *F = mod.createFunction("main");
 
-  inputsH.randomize(-1, 1, mod.getPRNG());
-  outputsH.randomize(-1, 1, mod.getPRNG());
+  auto *A = mod.createPlaceholder(ElemKind::FloatTy, {1, numOutputElem / 2},
+                                  "A", false);
+  bindings.allocate(A)->getHandle() = {1, 2};
+  auto *B = mod.createPlaceholder(ElemKind::FloatTy, {1, numOutputElem / 2},
+                                  "B", false);
+  bindings.allocate(B)->getHandle() = {3, 4};
 
-  performGradCheck(EE_, bindings, result, A, Exp, &inputs, &outputs, 0.001,
-                   0.01);
+  Node *O = F->createConcat("concat", {A, B}, 1);
+  Tensor expected(ElemKind::FloatTy, {1, numOutputElem / 2});
+  expected.getHandle() = {3, 4};
+  EXPECT_TRUE(expected.isEqual(*computeGradient(EE_, bindings, B, O)));
+}
+
+TEST_P(InterpreterGrad, gradientCheckAddLeft) {
+  PlaceholderBindings bindings;
+  size_t numOutputElem = 2;
+
+  auto &mod = EE_.getModule();
+  Function *F = mod.createFunction("main");
+
+  auto *A =
+      mod.createPlaceholder(ElemKind::FloatTy, {1, numOutputElem}, "A", false);
+  bindings.allocate(A)->getHandle() = {1, 2};
+  auto *B =
+      mod.createPlaceholder(ElemKind::FloatTy, {1, numOutputElem}, "B", false);
+  bindings.allocate(B)->getHandle() = {3, 4};
+
+  Node *O = F->createAdd("add", A, B);
+  // O = { { a1 + b1 }, { a2 + b2 } }
```

Comment:
We have similar but different comment in the actual grad computing code:
https://github.com/pytorch/glow/blob/2793672d9368f47690208acc3c8933944136cef9/lib/Optimizer/Lower.cpp#L47

Reply:
Thanks! I'll look into that and let me know if the style of this comment should be unified.